### ISSUE TITLE
Q: Un-aligned write

### ISSUE THREAD

**Issue by mslijepc** [Minres/SystemC-Components#45]
Hi Eyck, 
not sure if there is a bug in this line:

https://github.com/Minres/SystemC-Components/blob/7d8a0173b219d684c80ba268086b31027d4a7402/src/bus_interfaces/axi/pin/axi4_initiator.h#L174C13-L174C13

In case we have e.g. `32B` wide bus, we want to have single `4B` write, with an offset `30`,  `strb` will be `all zeros`. 
How this write should be handled?

**Comment by eyck** [comment]
I'll have a look...

**Comment by mslijepc** [comment]
and also, 'data' will be all zeros. Not sure, by the spec, if this should be done as burst?

**Comment by eyck** [comment]
Hmm, I connot reproduce the issue although I tried hard ( found an issue with the second beat but not the first one) . This is what I get:
![grafik](https://github.com/Minres/SystemC-Components/assets/437271/1c9467d8-5388-4400-92a9-688586c0764e)
Did you attach an axi4_extension to the incoming transaction?

**Comment by mslijepc** [comment]
Yes, axi4_extension was attached. How do you reproduce error, can you manually set address and len?

**Comment by eyck** [comment]
Address and length are taken from the tlm_generic_payload so this way you can control it
